### PR TITLE
Deflake wallet-connection tests

### PR DESCRIPTION
**Pull request by samsiegart**
refs: https://github.com/Agoric/agoric-sdk/issues/4248

### CODE REVIEW

File: packages/web-components/src/AgoricWalletConnection.js [hunk 1]
```diff
@@ -122,13 +122,15 @@ export const makeAgoricWalletConnection = (makeCapTP = defaultMakeCapTP) =>
           },
         });
         this.dispatchEvent(ev);
-        this.requestUpdate();
+        if (requestUpdate) {
+          this.requestUpdate();
```
Comment (samsiegart) [inline packages/web-components/src/AgoricWalletConnection.js:126]: Looks like removing this altogether breaks things, but it works fine removing it from `firstUpdated`.

File: packages/web-components/test/agoric-wallet-connection.test.js [hunk 2]
```diff
@@ -146,45 +162,40 @@ describe('AgoricWalletConnection', () => {
     it('goes to admin state once connected', async () => {
       iframeOnMessage('http://localhost:8000');
 
-      // Connecting happens instantly with the mock socket,
-      // just need to let the event loop run once.
-      await new Promise(resolve => setTimeout(resolve, 10));
+      await createStatePromise('bridged');
+      await nextFrame();
+      await nextFrame();
```
Comment (samsiegart) [inline packages/web-components/test/agoric-wallet-connection.test.js]: These `await nextFrame()`s are still needed unfortunately. If I only have 1, it's flaky, but using 2 passed 50 times in a row :/

File: packages/web-components/test/agoric-wallet-connection.test.js [hunk 3]
```diff
@@ -146,45 +162,40 @@ describe('AgoricWalletConnection', () => {
...
+
       expect(el.state).to.equal('bridged');
     });
 
     it('lets the websocket dispatch messages through capTP', async () => {
       iframeOnMessage('http://localhost:8000');
-      await new Promise(resolve => setTimeout(resolve, 10));
```
Comment (samsiegart) [inline packages/web-components/test/agoric-wallet-connection.test.js]: These weren't needed

**Comment by samsiegart** [comment]
Looks like this doesn't pass in CI, will take a look

File: packages/web-components/src/AgoricWalletConnection.js (hunk 1 above)
Comment (michaelfig) [reply packages/web-components/src/AgoricWalletConnection.js:126]: Great!  Thanks for fixing this.

**Review by michaelfig — COMMENTED** [review]
Looks reasonable.  I'll rereview and approve when CI passes.

**Comment by erights** [comment]
This version has a different failure in agoric-wallet-connection.test.js , as seen at https://github.com/Agoric/agoric-sdk/runs/4751043629?check_suite_focus=true

Unlike the problem this one fixes, this problem occurs under CI, so I'll rebase my work back to master for now.

**Comment by samsiegart** [comment]
> Looks reasonable. I'll rereview and approve when CI passes.

I changed it to use 100ms delays instead of `await nextFrame()` and CI passes now, multiple runs didn't show any flakes locally either, PTAL

**Review by michaelfig — APPROVED** [review]
LGTM!  Way to make lemonade. :)

**Pull request merged**